### PR TITLE
Vectorize Scale16X16To8X8

### PR DESCRIPTION
**Pull request by tkp1n**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp/pulls) open
- [x] I have verified that I am following matches the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description

In the context of #1476 I looked for low hanging fruit on the hot path of the JPEG encoding pipeline.
Using a benchmark (encoding a 4K image) with the ETW Profiler attached, I've noticed that `Block8x8F.Scale16X16To8X8` is responsible for 9.8% of the total time taken. I've noticed that the method is the optimal candidate for vectorization (AVX2) and went for it. 

* The benchmarks below show a **12x improvement** in `Block8x8F.Scale16X16To8X8` 🚀
* This reduces the total encoding time of my sample image by **~9%** from 475.9 ms to 434.1 ms

### Benchmarks

``` ini

BenchmarkDotNet=v0.12.1, OS=Windows 10.0.19042
Intel Core i7-9750H CPU 2.60GHz, 1 CPU, 12 logical and 6 physical cores
.NET Core SDK=5.0.102
  [Host] : .NET Core 3.1.11 (CoreCLR 4.700.20.56602, CoreFX 4.700.20.56604), X64 RyuJIT
  main   : .NET Core 3.1.11 (CoreCLR 4.700.20.56602, CoreFX 4.700.20.56604), X64 RyuJIT
  PR     : .NET Core 3.1.11 (CoreCLR 4.700.20.56602, CoreFX 4.700.20.56604), X64 RyuJIT

Runtime=.NET Core 3.1  

```
|          Method |  Job |      Mean |    Error |   StdDev | Ratio | Gen 0 | Gen 1 | Gen 2 | Allocated | Allocated native memory | Native memory leak |
|---------------- |----- |----------:|---------:|---------:|------:|------:|------:|------:|----------:|------------------------:|-------------------:|
| Scale16X16To8X8 | main | 189.83 ns | 2.875 ns | 2.689 ns |  1.00 |     - |     - |     - |         - |                       - |                  - |
| Scale16X16To8X8 |   PR |  14.97 ns | 0.179 ns | 0.159 ns |  0.08 |     - |     - |     - |         - |                       - |                  - |

### CODE REVIEW

**Comment by JimBobSquarePants** [comment]
Ah fantastic @tkp1n !!

I had a feeling this might be a bottle neck... I wonder, do you have screenshots of your trace we could use to identify further points of interest?

File: src/ImageSharp/Formats/Jpeg/Components/Block8x8F.cs [hunk 1]
```diff
@@ -485,6 +485,57 @@ public static unsafe void DequantizeBlock(Block8x8F* blockPtr, Block8x8F* qtPtr,
...
+            ref Vector256<float> destRef = ref Unsafe.As<Block8x8F, Vector256<float>>(ref destination);
+
+            for (int i = 0; i < 8; i++)
+            {
+                Vector256<float> a = in1;
+                Vector256<float> b = Unsafe.Add(ref in1, 1);
+                Vector256<float> c = in2;
+                Vector256<float> d = Unsafe.Add(ref in2, 1);
+
+                Vector256<float> calc1 = Avx.Shuffle(a, c, 0b10_00_10_00);
```
Comment (JimBobSquarePants) [inline src/ImageSharp/Formats/Jpeg/Components/Block8x8F.cs:519]: I wish there was a way to produce constants via `SimdUtils.Shuffle.MmShuffle()` in a way that could be inlined.

**Review by JimBobSquarePants — APPROVED** [review]
12X... Can't really ask for more than that. Fantastic stuff!

**Comment by antonfirsov** [comment]
9%+ with such a little addition, very nice!

**Pull request merged**